File: 2021/day1.py
#!/usr/bin/env python3

WINDOW_SIZE = 3

def main():
	increases = 0
	with open('day1_input', 'r', encoding='ascii') as f:
		prev = []
		prev_sum = None
		for line in f:
			cur = int(line.rstrip())
			if len(prev) < WINDOW_SIZE - 1:
				prev.append(cur)
				continue

			cur_sum = sum(prev) + cur
			if prev_sum is not None and cur_sum > prev_sum:
				increases += 1

			prev.pop(0)
			prev.append(cur)
			prev_sum = cur_sum
	print(increases)

if __name__ == '__main__':
	main()

File: 2021/day1.exs
inp = File.stream!("day1_input") |> Stream.map(&String.trim_trailing(&1)) |> Stream.map(&String.to_integer(&1))
increases = Enum.reduce inp, {0, nil}, fn
	line, {0, nil} -> {0, line}
	line, {increases, prev} when line > prev -> {increases + 1, line}
	line, {increases, prev} when line <= prev -> {increases, line}
end
IO.inspect(elem(increases, 0))
